### PR TITLE
Add human intervention mechanism and eval_real script to evaluate policy on the robot

### PR DESCRIPTION
**Pull request by michel-aractingi**
## What this does
Related to the human intervention category in issue #504 . What this does:
 - Adds an `eval_robot.py` script to run rollouts of the policy on the real robot. 
 - Adds the ability to use the keyboard to pause the policy actions. 
 - Adds the ability to allow the user to take over the policy by using the leader arm to control the follower through teleoperation. 

## How it was tested:
In the main of the `eval_robot.py` we run the `eval_policy` function that runs the robot's action in a fixed command (in the future this will be the policy actions). Once the `space` is pressed the actions are paused and message is written for the human to prepare the leader in a similar pose to take over. Once another `space` is pressed the teleoperation begins. The user can press the right arrow to move on to the next rollout or the left arrow to rerecord the episode. 

you can test it yourself by:
```
python lerobot/scripts/eval_robot.py --robot-path lerobot/configs/robot/koch.yaml
```

## TODOs
- [ ] fix the timing when the policy is paused so that the time is not counted with the control time. 
- [ ] add reward classifier from #528 to label the rewards of the rollouts. 
- [ ] add mechanism to classify success
- [ ] test exploration with policy

### CODE REVIEW

**Comment by michel-aractingi** [comment]
cc @ChorntonYoel

File: lerobot/scripts/eval_robot.py [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+                "episode_ix": i,
+                "sum_reward": sum_reward,
+                "max_reward": max_reward,
+                "pc_success": float(np.nanmean(successes[:n_episodes]) * 100),
+            }
+            for i, (sum_reward, max_reward) in enumerate(
+                zip(
+                    sum_rewards[:n_episodes],
+                    max_rewards[:n_episodes],
+                    successes[:n_episodes]
```
Comment (ChorntonYoel) [inline lerobot/scripts/eval_robot.py:240]:
```suggestion
```

File: lerobot/scripts/eval_robot.py [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+                )
+            )
+        ],
+        "aggregated": {
+            "avg_sum_reward": float(np.nanmean(sum_rewards[:n_episodes])),
+            "avg_max_reward": float(np.nanmean(max_rewards[:n_episodes])),
+            "pc_success": float(np.mean(successes) * 100),
```
Comment (ChorntonYoel) [inline lerobot/scripts/eval_robot.py:247]:
```suggestion
            "avg_sum_reward": float(np.nanmean(torch.cat(sum_rewards[:n_episodes]))),
            "avg_max_reward": float(np.nanmean(torch.cat(max_rewards[:n_episodes]))),
            "pc_success": float(np.nanmean(torch.cat(successes[:n_episodes])) * 100),
```
Should double check that it's computing what you want, but otherwise it's throwing errors for me

File: lerobot/scripts/eval_robot.py [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+        max_rewards.append(max(rollout_data['next.reward']))
+        successes.append(rollout_data["next.success"][-1])
+        
+    info = {
+        "per_episode": [
+            {
+                "episode_ix": i,
+                "sum_reward": sum_reward,
+                "max_reward": max_reward,
+                "pc_success": float(np.nanmean(successes[:n_episodes]) * 100),
```
Comment (ChorntonYoel) [inline lerobot/scripts/eval_robot.py:234]: this one is computed both in aggregated and per_episode, is this what we want?

File: lerobot/scripts/eval_robot.py [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+                # if second space press then the user is ready to start intervention  
+                if not events["pause_policy"]:
+                    print("Space key pressed. Human intervention required.\n" \
+                          "Place the leader in similar pose to the follower and press space again.")
+                    events["pause_policy"] = True
+                    log_say("Human intervention stage. Get ready to take over.", play_sounds=True)
+                else:
+                    events["human_intervention_step"] = True
+                    print("Space key pressed. Human intervention starting.")
+                    log_say("Starting human intervention.", play_sounds=True)
```
Comment (ChorntonYoel) [inline lerobot/scripts/eval_robot.py:306]: those are nice. Didn't use for the reward assignment, but we probably should

File: lerobot/scripts/eval_robot.py (hunk 3 above)
Comment (ChorntonYoel) [inline lerobot/scripts/eval_robot.py:234]:
```suggestion
                "pc_success": float(np.nanmean(torch.cat(successes[:n_episodes])) * 100),
```

**Review by ChorntonYoel — COMMENTED** [review]
Looks good thanks! I have a couple comments for the episode data aggregation, it wasn't working for me so I think it needs a few tweaks.
Policy is not used yet so I just tested without rn but seems to be working quite well. 

I like the left/right arrow feature to rerecord/finish quicker

File: lerobot/scripts/eval_robot.py (hunk 3 above)
Comment (michel-aractingi) [reply lerobot/scripts/eval_robot.py]: yes one is aggregated over all rollouts (episodes) and the other is stats per episode
